median.cc

```diff
@@ -0,0 +1,289 @@
+// Copyright 2015 Rocana, Inc.
```

Comment:
I assume this is the minimum we could borrow from Impala's aggregate-functions.cc to parametrize the maximum number of samples?